mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -179,16 +211,13 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
       inputName: String,
       explodeCols: Seq[String]): Dataset[_] = {
     require(explodeCols.size == 2, "explodeCols must be two strings.")
-    val vectorToMap = udf((x: Vector) => x.asBreeze.iterator.toMap,
-      MapType(DataTypes.IntegerType, DataTypes.DoubleType))
     val modelDataset: DataFrame = if (!dataset.columns.contains($(outputCol))) {
       transform(dataset)
     } else {
       dataset.toDF()
     }
     modelDataset.select(
-      struct(col("*")).as(inputName),
-      explode(vectorToMap(col($(outputCol)))).as(explodeCols))
+      struct(col("*")).as(inputName), posexplode(col($(outputCol))).as(explodeCols))
```

Comment:
Yes, I copied your code to `BucketedRandomProjectionLSHSuite.scala` and it runs fine for me with the following output:
```
+--------------------+--------------------+------------------+
|            datasetA|            datasetB|           distCol|
+--------------------+--------------------+------------------+
|[(50,[0],[-2.0]),...|[(50,[0],[-1.0]),...|               1.0|
|[(50,[4],[-1.0]),...|[(50,[23],[-1.0])...|1.4142135623730951|
|[(50,[5],[-1.0]),...|[(50,[32],[-1.0])...|1.4142135623730951|
|[(50,[7],[1.0]),W...|[(50,[47],[1.0]),...|1.4142135623730951|
|[(50,[7],[2.0]),W...|[(50,[26],[-2.0])...|2.8284271247461903|
|[(50,[8],[-2.0]),...|[(50,[1],[-1.0]),...|  2.23606797749979|
|[(50,[8],[-1.0]),...|[(50,[23],[-2.0])...|  2.23606797749979|
|[(50,[10],[-1.0])...|[(50,[7],[2.0]),W...|  2.23606797749979|
|[(50,[10],[-1.0])...|[(50,[13],[2.0]),...|  2.23606797749979|
|[(50,[11],[-1.0])...|[(50,[39],[2.0]),...|  2.23606797749979|
|[(50,[12],[-2.0])...|[(50,[28],[1.0]),...|  2.23606797749979|
|[(50,[12],[-2.0])...|[(50,[29],[-1.0])...|  2.23606797749979|
|[(50,[13],[1.0]),...|[(50,[2],[-2.0]),...|  2.23606797749979|
|[(50,[14],[1.0]),...|[(50,[33],[2.0]),...|  2.23606797749979|
|[(50,[14],[2.0]),...|[(50,[28],[2.0]),...|2.8284271247461903|
|[(50,[15],[-1.0])...|[(50,[38],[-1.0])...|1.4142135623730951|
|[(50,[18],[1.0]),...|[(50,[8],[-1.0]),...|1.4142135623730951|
|[(50,[18],[1.0]),...|[(50,[12],[-2.0])...|  2.23606797749979|
|[(50,[18],[2.0]),...|[(50,[43],[1.0]),...|  2.23606797749979|
|[(50,[20],[1.0]),...|[(50,[25],[-1.0])...|1.4142135623730951|
+--------------------+--------------------+------------------+
only showing top 20 rows

Let me see if the test can pass jenkins or not.
```